StackExchange.Redis/StackExchange/Redis/SocketManager.cs

```diff
@@ -371,6 +396,28 @@ private void EndConnectImpl(IAsyncResult ar, ConnectionMultiplexer multiplexer,
             }
         }
 
+#if CORE_CLR
+        private async Task<EndPoint> ResolveWorkingEndpointAsync(DnsEndPoint dnsEndPoint, AddressFamily family, SocketType socketType, ProtocolType protocol)
+        {
+            var hostAddresses = await Dns.GetHostAddressesAsync(dnsEndPoint.Host);
+            foreach (var address in hostAddresses)
+            {
+                try
+                {
+                    var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+                    var endpoint = new IPEndPoint(address, dnsEndPoint.Port);
+                    await socket.ConnectAsync(endpoint);

```

Comment:
This could take long.  Could it be possible that one IP endpoint causes timeout, while in the original windows code, the Dns endpoint still works?